autopilot/contractor.go

```diff
@@ -1201,20 +1201,20 @@ func (c *contractor) renewFundingEstimate(ctx context.Context, ci contractInfo,
 	return cappedEstimatedCost, nil
 }
 
-func (c *contractor) calculateMinScore(ctx context.Context, scoredHosts map[types.PublicKey]scoredHost, numContracts uint64) (float64, error) {
+func (c *contractor) calculateMinScore(ctx context.Context, candidates []scoredHost, numContracts uint64) (float64, error) {
 	// return early if there's no hosts
-	if len(scoredHosts) == 0 {
-		c.logger.Warn("min host score is set to the smallest non-zero float because there are no hosts that passed scoring")
+	if len(candidates) == 0 {
+		c.logger.Warn("min host score is set to the smallest non-zero float because there are no candidate hosts")
 		return math.SmallestNonzeroFloat64, nil
 	}
 
 	// do multiple rounds to select the lowest score
 	var lowestScores []float64
 	for r := 0; r < 5; r++ {
 		lowestScore := math.MaxFloat64
-		for _, host := range randSelectHosts(scoredHosts, int(numContracts)+50) { // buffer
-			if scoredHosts[host.PublicKey].score < lowestScore {
-				lowestScore = scoredHosts[host.PublicKey].score
+		for _, host := range scoredHosts(candidates).randSelectByScore(int(numContracts) + 50) { // buffer
```

Comment:
I'm not sure it's worth adding a new type to get from `randSelectByScore(candidates, int(numContracts) + 50)` to `scoredHosts(candidates).randSelectByScore(int(numContracts) + 50)`. 
It would make a bit more sense if we were actually using the `scoredHosts` type to pass the hosts around instead of `[]scoredHost` but making parsing the required step for discovering a method on a type that could otherwise be a function seems a little less obvious.

Reply:
got rid of parsing